cpp/include/cudf/round_robin.hpp

```diff
@@ -29,8 +29,36 @@ namespace experimental {
  * Returns a new table with rows re-arranged into partition groups and
  * a vector of row offsets to the start of each partition in the output table.
  * Rows are assigned partitions based on their row index in the table,
- * based on the following formula:
- * partition = (row_index + start_partition) % num_partitions
+ * in a round robin fashion.
+ *
+ * @throws cudf::logic_error if num_partitions <= 1 
+ * @throws cudf::logic_error if start_partition >= num_partitions.
+ *
+ * A good analogy for the algorithm is dealing out cards:
+ *
+ *  1. The deck of cards is represented as the rows in the table.
+ *  2. The number of partitions is the number of players being dealt cards.
+ *  3. the start_partition indicates which player starts getting cards first.
+ *
+ * The algorithm has two outcomes:
+ *
+ *  (a) Another deck of cards formed by stacking each 
+ *      player's cards back into a deck again, 
+ *      preserving the order of cards dealt to each player, 
+ *      starting with player 0.
+ *  (b) A vector into the output deck indicating where a player's cards start.
+ *
+ * A player's deck (partition) is the range of cards starting 
+ * at the corresponding offset and ending at the next player's 
+ * starting offset or the last card in the deck if it's the last player.
+ *
+ * When num_partitions > nrows, we have more players than decks. 
+ * We start dealing to the first indicated player and continuing 
+ * around the players until we run out of cards before we run out of players. 
+ * Players that did not get any cards are represented by
+ * offset[i] == offset[i+1] or
```

Comment:
To me it looks like this way of representing players w/o cards is convenient for the implementation (makes inclusive_scan directly applicable), but not for the user. AFAIK, the check for this would take at least three conditions, carefully implemented to avoid out-of-bound access.
Are there some other reason for the current scheme? 
If not, I would prefer something like simple "offset[i] == table.num_rows()" (or -1), which can be checked very easily.

Reply:
This is copied verbatim from @jlowe comments on previous round-robin PR. @jlowe, could you please address @vuule 's comments?

Reply:
The problem with using `offset[i] == table.num_rows()` is that it changes the ending offset of the previous partition.  Each offset serves two roles: where the current partition starts and where the previous one ends.

The other reason to use this convention is because that's what `split` expects.  Partition is almost always followed by splitting so the partitions can be shipped off to different downstream tasks.

Reply:
Okay, what's the action item for me, here? Can this thread be closed (resolved), then?

Reply:
I do not see any further action item for you, and I think this thread can be resolved, assuming @vuule agrees.

Reply:
Thanks for the explanation, I suspected there was more to it.
I don't have issues with the current documentation.
Resolving the thread.

Reply:
Thanks for the explanation, I suspected there was more to it.
I don't have issues with the current documentation.
Resolving the thread.